salt/metalk8s/addons/nginx-ingress-control-plane/control-plane-ip.sls

```diff
@@ -0,0 +1,28 @@
+{# This whole block is used to "know" the Ingress external IP used by Dex.
+  It will be removed once we can have a known LoadBalancer IP for Ingress. #}
+{% if '_errors' in pillar.metalk8s.nodes %}
+  {# Assume this is the bootstrap Node and we haven't an apiserver yet #}
+  {%- set bootstrap_id = grains.id %}
+{%- elif pillar.metalk8s.nodes | length <= 1 %}
+  {# Only one node (or even, zero) can/should only happen during bootstrap #}
+  {%- set bootstrap_id = grains.id %}
+{%- else %}
+  {%- set bootstrap_nodes = salt.metalk8s.minions_by_role('bootstrap') %}
+  {%- if bootstrap_nodes %}
+    {%- set bootstrap_id = bootstrap_nodes | first %}
+  {%- else %}
+    {{ raise('Missing bootstrap node') }}
+  {%- endif %}
+{%- endif %}
+
+{%- if bootstrap_id is none %}
+  {{ raise('Missing bootstrap Node in pillar, cannot proceed.') }}
+{%- elif bootstrap_id == grains.id %}
+  {%- set bootstrap_control_plane_ip = grains.metalk8s.control_plane_ip %}
+{%- else %}
+  {%- set bootstrap_control_plane_ip = salt['mine.get'](bootstrap_id,
+                                                        'control_plane_ip') %}
```

Comment:
Here you have an issue you will get a dict contains control_plane_ip of each and every node so if you want to do it you need to take only the bootstrap one so

```
  {%- set bootstrap_control_plane_ip = salt['mine.get'](bootstrap_id,
                                                        'control_plane_ip')[bootstrap_id] %}
```